Sources/JWTKit/JWTSerializer.swift

```diff
@@ -1,19 +1,32 @@
+import Foundation
+import X509
+
 struct JWTSerializer {
     func sign(
         _ payload: some JWTPayload,
         using signer: JWTSigner,
         typ: String = "JWT",
         kid: JWKIdentifier? = nil,
         cty: String? = nil,
+        x5c: [String]? = nil,
         jsonEncoder: any JWTJSONEncoder
-    ) throws -> String {
+    ) async throws -> String {
         // encode header, copying header struct to mutate alg
         var header = JWTHeader()
         header.kid = kid
         header.typ = typ
         header.cty = cty
         header.alg = signer.algorithm.name
 
+        if let x5c, !x5c.isEmpty {
+            let verifier = try X5CVerifier(rootCertificates: [x5c[0]])
+            try await verifier.verifyChain(certificates: x5c)
```

Comment:
We may want to add a flag to disable verification, for testing etc. But I'm not sure if this is explicitly disallowed